ifsbench/job.py

```diff
@@ -107,206 +89,134 @@ class CpuDistribution(Enum):
     DISTRIBUTE_USER = auto()
     """Indicate that a different user-specified strategy should be used"""
 
+@dataclass
 class Job:
     """
-    Description of a parallel job's resource requirements
-
-    Provided with a CPU configuration (:data:`cpu_config`) and at least one of
-
-    * the total number of MPI tasks (:data:`tasks`)
-    * the number of nodes (:data:`nodes`) and the number of tasks per node
-      (:data:`tasks_per_node`)
-    * the number of nodes (:data:`nodes`) and the number of tasks per socket
-      (:data:`tasks_per_socket`)
-
-    this class specifies the resource requirements for a job.
-
-    The underlying idea is to specify as little as possible which is then passed on
-    to the relevant launch command but with the possibility to estimate those values
-    that can be derived unambigously from the specified values.
-
-    The relevant attributes (with the same names as the parameters to the constructor)
-    are only defined when the corresponding value has been specified explicitly (with
-    the exception of those that have a default value), i.e., accessing undefined
-    attributes will raise a :any:`AttributeError`. The corresponding ``get_`` methods
-    allow to derive the relevant values when it is unambigously possible, or raise an
-    :any:`AttributeError` otherwise.
-
-    Multi-threading can be specified by providing a value larger than 1 (the default)
-    for :data:`cpus_per_task`. Symmetric multi-threading (hyperthreading) can be
-    enabled with a value greater than 1 in :data:`threads_per_core`.
-
-    The desired pinning strategy can be specified with :data:`bind`.
+    Description of a parallel job setup.
 
     Parameters
```

Comment:
These are no longer parameters to the constructor, so should be converted to attribute docstrings